presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/SfmSketchStateFactory.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation;
+
+import com.facebook.presto.common.array.ObjectBigArray;
+import com.facebook.presto.operator.aggregation.noisyaggregation.sketch.SfmSketch;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+import com.facebook.presto.spi.function.GroupedAccumulatorState;
+import org.openjdk.jol.info.ClassLayout;
+
+import static java.util.Objects.requireNonNull;
+
+public class SfmSketchStateFactory
+        implements AccumulatorStateFactory<SfmSketchState>
+{
+    @Override
+    public SfmSketchState createSingleState()
+    {
+        return new SingleSfmSketchState();
+    }
+
+    @Override
+    public Class<? extends SfmSketchState> getSingleStateClass()
+    {
+        return SingleSfmSketchState.class;
+    }
+
+    @Override
+    public SfmSketchState createGroupedState()
+    {
+        return new GroupedSfmSketchState();
+    }
+
+    @Override
+    public Class<? extends SfmSketchState> getGroupedStateClass()
+    {
+        return GroupedSfmSketchState.class;
+    }
+
+    public static class GroupedSfmSketchState
+            implements SfmSketchState, GroupedAccumulatorState
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(GroupedSfmSketchState.class).instanceSize();
+
+        private final ObjectBigArray<SfmSketch> sketches = new ObjectBigArray<>();
+        private final ObjectBigArray<Double> epsilons = new ObjectBigArray<>();
+
+        private long retainedBytes;
+        private long groupId;
+
+        @Override
+        public final void setGroupId(long groupId)
+        {
+            this.groupId = groupId;
+        }
+
+        protected final long getGroupId()
+        {
+            return groupId;
+        }
+
+        @Override
+        public void ensureCapacity(long size)
+        {
+            epsilons.ensureCapacity(size);
+            sketches.ensureCapacity(size);
+        }
+
+        @Override
+        public SfmSketch getSketch()
+        {
+            return sketches.get(getGroupId());
+        }
+
+        @Override
+        public void setSketch(SfmSketch value)
+        {
+            requireNonNull(value, "value is null");
+            sketches.set(getGroupId(), value);
```

Comment:
That's correct. There used to be such a line here, but I deleted it, since it was a no-op. I can put it back, though, just in case anything ever changes in the future.